controlplane/pkg/nsmd/nsmd_monitor_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package nsmd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/networkservicemesh/networkservicemesh/sdk/monitor/connectionmonitor"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/monitor/remote"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/networkservicemesh/sdk/monitor"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+)
+
+const sendPeriod = time.Millisecond * 15
+const testingTime = time.Second / 4
+
+func TestNsmdMonitorShouldHandleServerShutdown(t *testing.T) {
+	port, stop := setupServer(t)
+	go func() {
+		<-time.After(testingTime / 2)
+		stop()
+	}()
+	err := startClient(t, testingTime, port)
+	if err == nil {
```

Comment:
Why don't you use `Expect(err).To(...)`?

Reply:
the test not using gomaga

Reply:
Why test isn't using gomega?

Reply:
thanks, fixed.